sphinx/application.py

```diff
@@ -13,9 +13,11 @@
 from collections.abc import Callable, Collection, Sequence  # NoQA: TCH003
 from io import StringIO
 from os import path
-from typing import IO, TYPE_CHECKING, Any, Literal
+from pathlib import Path  # NoQA: TCH003
```

Comment:
Not sure about this. The preference so far has been to suppress the autodoc warning, not the lint warning.
I don't have a strong preference on which is better but we should be consistent

Reply:
I chose this because it is already done in this file. In particular, for example, on line 13 (just three lines up from this).

I'm open to changing it.

Reply:
You're right, there are examples of both.

I think the team should take a decision on this, but it shouldn't block your PR. You're approach is locally consistent. That's good enough for now!

Reply:
Thank you. You can see that this change includes one `nitpick_ignore` addition.
I tried to avoid this with `import docutils.nodes` (to leverage the existing `nitpick_ignore` for `docutils.nodes.TextElement`) but there was a conflict as there is already an import of a module named `docutils`).